File: haskell/glue/src/Glue/Lib/Math/Const.hs
module Glue.Lib.Math.Const where

import Glue.Eval (Eval)
import Glue.IR (IR (..))
import Glue.Module (ModuleInfo, nativeModule)

const :: ModuleInfo Eval
const =
    nativeModule
        "ffi.math.const"
        [ ("e", Float (exp 1))
        , ("infinity", Float (1 / 0))
        , ("minus-infinity", Float ((-1) / 0))
        , ("pi", Float pi)
        , ("phi", Float ((1 + sqrt 5) / 2))
        ]

File: dart/glue/lib/src/lib/math/const.dart
import 'dart:math' as math;

import 'package:glue/src/ir.dart';
import 'package:glue/src/module.dart';

/// Math constants module
/// Mirrors Haskell Glue.Lib.Math.Const exactly

/// The const module containing mathematical constants
/// Mirrors Haskell Glue.Lib.Math.Const.const exactly
final ModuleInfo constModule = nativeModule('ffi.math.const', [
  ('e', IrFloat(math.e)),
  ('infinity', IrFloat(double.infinity)),
  ('minus-infinity', IrFloat(-double.infinity)),
  ('pi', IrFloat(math.pi)),
  ('phi', IrFloat((1 + math.sqrt(5)) / 2)),
  ('nan', IrFloat(double.nan)),
]);
